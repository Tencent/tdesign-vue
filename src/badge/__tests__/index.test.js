import { mount } from '@vue/test-utils';
import Badge from '@/src/badge/index.ts';

// every component needs four parts: props/events/slots/functions.
describe('Badge', () => {
  // test props api
  describe(':props', () => {
    it(':color', () => {
      const wrapper = mount({
        render() {
          return <Badge color="green">Tdesign</Badge>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':count', () => {
      const wrapper = mount({
        render() {
          return <Badge count={2}>Tdesign</Badge>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':dot', () => {
      const wrapper = mount({
        render() {
          return (
            <Badge dot={true} count={2}>
              Tdesign
            </Badge>
          );
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':maxCount', () => {
      const wrapper = mount({
        render() {
          return (
            <Badge maxCount={33} count={2}>
              Tdesign
            </Badge>
          );
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':count', () => {
      const wrapper = mount({
        render() {
          return (
            <Badge count="new" count={2}>
              Tdesign
            </Badge>
          );
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':size', () => {
      const wrapper = mount({
        render() {
          return (
            <Badge size="small" count={2}>
              Tdesign
            </Badge>
          );
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':shape', () => {
      const wrapper = mount({
        render() {
          return (
            <Badge shape="round" count={2}>
              Tdesign
            </Badge>
          );
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':showZero', () => {
      const wrapper = mount({
        render() {
          return (
            <Badge showZero={true} count={2}>
              Tdesign
            </Badge>
          );
        },
      });
      expect(wrapper).toMatchSnapshot();
    });

    it(':offset', () => {
      const wrapper = mount({
        render() {
          return (
            <Badge offset={[10, 10]} count={2}>
              Tdesign
            </Badge>
          );
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });

  // test slots
  describe('<slot>', () => {
    it('default', () => {
      const wrapper = mount({
        render() {
          return <Badge count={2}>Tdesign</Badge>;
        },
      });
      expect(wrapper).toMatchSnapshot();
    });
  });
});
