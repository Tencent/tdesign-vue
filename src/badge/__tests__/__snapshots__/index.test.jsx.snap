// Vitest Snapshot v1

exports[`Badge > :props > :color 1`] = `
VueWrapper {
  "_emitted": {
    "hook:mounted": [
      [],
    ],
  },
  "_emittedByOrder": [
    {
      "args": [],
      "name": "hook:mounted",
    },
  ],
  "isFunctionalComponent": undefined,
}
`;

exports[`Badge > :props > :count 1`] = `
VueWrapper {
  "_emitted": {
    "hook:mounted": [
      [],
    ],
  },
  "_emittedByOrder": [
    {
      "args": [],
      "name": "hook:mounted",
    },
  ],
  "isFunctionalComponent": undefined,
}
`;

exports[`Badge > :props > :count 2`] = `
VueWrapper {
  "_emitted": {
    "hook:mounted": [
      [],
    ],
  },
  "_emittedByOrder": [
    {
      "args": [],
      "name": "hook:mounted",
    },
  ],
  "isFunctionalComponent": undefined,
}
`;

exports[`Badge > :props > :dot 1`] = `
VueWrapper {
  "_emitted": {
    "hook:mounted": [
      [],
    ],
  },
  "_emittedByOrder": [
    {
      "args": [],
      "name": "hook:mounted",
    },
  ],
  "isFunctionalComponent": undefined,
}
`;

exports[`Badge > :props > :maxCount 1`] = `
VueWrapper {
  "_emitted": {
    "hook:mounted": [
      [],
    ],
  },
  "_emittedByOrder": [
    {
      "args": [],
      "name": "hook:mounted",
    },
  ],
  "isFunctionalComponent": undefined,
}
`;

exports[`Badge > :props > :offset 1`] = `
VueWrapper {
  "_emitted": {
    "hook:mounted": [
      [],
    ],
  },
  "_emittedByOrder": [
    {
      "args": [],
      "name": "hook:mounted",
    },
  ],
  "isFunctionalComponent": undefined,
}
`;

exports[`Badge > :props > :shape 1`] = `
VueWrapper {
  "_emitted": {
    "hook:mounted": [
      [],
    ],
  },
  "_emittedByOrder": [
    {
      "args": [],
      "name": "hook:mounted",
    },
  ],
  "isFunctionalComponent": undefined,
}
`;

exports[`Badge > :props > :showZero 1`] = `
VueWrapper {
  "_emitted": {
    "hook:mounted": [
      [],
    ],
  },
  "_emittedByOrder": [
    {
      "args": [],
      "name": "hook:mounted",
    },
  ],
  "isFunctionalComponent": undefined,
}
`;

exports[`Badge > :props > :size 1`] = `
VueWrapper {
  "_emitted": {
    "hook:mounted": [
      [],
    ],
  },
  "_emittedByOrder": [
    {
      "args": [],
      "name": "hook:mounted",
    },
  ],
  "isFunctionalComponent": undefined,
}
`;

exports[`Badge > <slot> > default 1`] = `
VueWrapper {
  "_emitted": {
    "hook:mounted": [
      [],
    ],
  },
  "_emittedByOrder": [
    {
      "args": [],
      "name": "hook:mounted",
    },
  ],
  "isFunctionalComponent": undefined,
}
`;
